### ISSUE TITLE
JSON produces ComputeError(ErrString("expected list/array in json value, got str"))

### ISSUE THREAD

**Issue by ghost** [pola-rs/polars#8470]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

Parsing a folder of JSON's I encounted a seemingly valid JSON file that Polars struggles with. I cannot identify what the issue is and Pandas reads the file without issue.

These are two example files (I had to rename the extension to .txt for Github upload to work):
[broken.txt](https://github.com/pola-rs/polars/files/11309421/broken.txt)
[works.txt](https://github.com/pola-rs/polars/files/11309422/works.txt)

### Reproducible example

```python
import polars as pl
import pandas as pd

pl.scan_ndjson(r"works.txt").collect()
# This creates a dataframe with 4 columns of structs

pd.read_json(r"broken.txt")
# This creates a dataframe with toplevel keys as columns and 2nd level keys as rows

pl.scan_ndjson(r"broken.txt").collect()
# This output PanicException: inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))

pl.read_ndjson(r"broken.txt")
# This output PanicException: inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))
```

### Expected behavior

I expect that the `broken.txt` file loads just as the `works.txt` file as both are valid JSON (according to https://jsonlint.com/).

### Installed versions

<details>

```
---Version info---
Polars: 0.17.7
Index type: UInt32
Platform: Windows-10-10.0.14393-SP0
Python: 3.11.3 | packaged by conda-forge | (main, Apr  6 2023, 08:50:54) [MSC v.1934 64 bit (AMD64)]
---Optional dependencies---
numpy: 1.24.3
pandas: 2.0.0
pyarrow: 11.0.0
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: <not installed>
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

</details>

**Comment by cmdlineluser** [comment]
Neither of the files load for me:
```
# RuntimeError: BindingsError: "Not yet implemented: read an Array from a non-Array JSON"
pl.read_json("broken.json")
```

```
# RuntimeError: BindingsError: "Not yet implemented: read an Array from a non-Array JSON"
pl.read_json("working.json")
```

```
# RuntimeError: BindingsError: "External format error: InvalidEOF"
pl.read_ndjson("broken.json")
```

```
# RuntimeError: BindingsError: "External format error: InvalidEOF"
pl.read_ndjson("working.json")
```

It does load using the [`.str.json_extract`](https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.str.json_extract.html#polars.Expr.str.json_extract) api.
```
from pathlib import Path

pl.DataFrame(
   {"json": Path("broken.json").read_text()}
).with_columns(pl.col("json").str.json_extract()).unnest("json")

# shape: (1, 4)
# ┌───────────────────────────────────┬───────────────────────────────────┬────────────────────────────┬───────────────────┐
# │ requestParameters                 ┆ plan                              ┆ debugOutput                ┆ elevationMetadata │
# │ ---                               ┆ ---                               ┆ ---                        ┆ ---               │
# │ struct[10]                        ┆ struct[4]                         ┆ struct[6]                  ┆ struct[2]         │
# ╞═══════════════════════════════════╪═══════════════════════════════════╪════════════════════════════╪═══════════════════╡
# │ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
# └───────────────────────────────────┴───────────────────────────────────┴────────────────────────────┴───────────────────┘
```

**Comment by ba-tno** [comment]
Thanks for the suggestion on the [`.str.json_extract`](https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.str.json_extract.html#polars.Expr.str.json_extract) api!

`.read_json`  expectedly does not work as it wants an array of JSON objects, i.e. `[ { "object": {"key": "value}, ... } ]` instead of `{ "object": {"key": "value}, ... }`.

The `InvalidEOF` is strange, I cannot reproduce it on my machine. I created a new conda environment with:

```
conda env create -n polarsdemo
mamba install python
pip install polars
pip install requests
```

Then ran this script

```py
import polars as pl
from polars.exceptions import PolarsPanicError
import requests
import os


os.environ["RUST_BACKTRACE"] = "full"

urls = {
    "broken": "https://github.com/pola-rs/polars/files/11309421/broken.txt",
    "works": "https://github.com/pola-rs/polars/files/11309422/works.txt",
}

for fname, url in urls.items():
    print(f"---- {fname} ----")
    r = requests.get(url, allow_redirects=True)
    open(f"{fname}.json", "wb").write(r.content)
    try:
        df = pl.read_ndjson(f"{fname}.json")
    except PolarsPanicError as ppe:
        print(ppe)
    else:
        print(df)

pl.show_versions()
```

and got this output:

```txt
---- broken ----
thread '<unnamed>' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:267:53
stack backtrace:
   0:     0x7ffd177b967d - ffi_select_with_compiled_path
   1:     0x7ffd15c8de7b - BrotliDecoderVersion
   2:     0x7ffd177a8eea - ffi_select_with_compiled_path
   3:     0x7ffd177bbb3b - ffi_select_with_compiled_path
   4:     0x7ffd177bb739 - ffi_select_with_compiled_path
   5:     0x7ffd177bc689 - ffi_select_with_compiled_path
   6:     0x7ffd177bc0e6 - ffi_select_with_compiled_path
   7:     0x7ffd177bc039 - ffi_select_with_compiled_path
   8:     0x7ffd177bc024 - ffi_select_with_compiled_path
   9:     0x7ffd1787abb5 - ffi_select_with_compiled_path
  10:     0x7ffd1787af86 - ffi_select_with_compiled_path
  11:     0x7ffd16b3692f - ffi_select_with_compiled_path
  12:     0x7ffd16b32203 - ffi_select_with_compiled_path
  13:     0x7ffd16b319fd - ffi_select_with_compiled_path
  14:     0x7ffd16b3d9a6 - ffi_select_with_compiled_path
  15:     0x7ffd1796ff21 - ffi_select_with_compiled_path
  16:     0x7ffd1766b860 - ffi_select_with_compiled_path
  17:     0x7ffd1766b37b - ffi_select_with_compiled_path
  18:     0x7ffd177b86ec - ffi_select_with_compiled_path
  19:     0x7ffdc43f7604 - BaseThreadInitThunk
  20:     0x7ffdc4ba26a1 - RtlUserThreadStart
inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))
---- works ----
shape: (1, 4)
┌───────────────────────────────────┬───────────────────────────────────┬────────────────────────────┬───────────────────┐
│ requestParameters                 ┆ plan                              ┆ debugOutput                ┆ elevationMetadata │
│ ---                               ┆ ---                               ┆ ---                        ┆ ---               │
│ struct[10]                        ┆ struct[4]                         ┆ struct[6]                  ┆ struct[2]         │
╞═══════════════════════════════════╪═══════════════════════════════════╪════════════════════════════╪═══════════════════╡
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682576400000,{"Origin",5.8045,… ┆ {99,327,[326],2,428,false} ┆ {47.694914,false} │
└───────────────────────────────────┴───────────────────────────────────┴────────────────────────────┴───────────────────┘
---Version info---
Polars: 0.17.7
Index type: UInt32
Platform: Windows-10-10.0.19045-SP0
Python: 3.11.3 | packaged by conda-forge | (main, Apr  6 2023, 08:50:54) [MSC v.1934 64 bit (AMD64)]
---Optional dependencies---
numpy: <not installed>
pandas: <not installed>
pyarrow: <not installed>
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: <not installed>
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

**Comment by ba-tno** [comment]
Although slightly unwieldy, surprisingly the `str.json_extract` isn't slow either (I thought the `read_text()` might slow it down). I tried the same code with the `works.json` and the other methods took more or less the same time.

```py
import polars as pl
from pathlib import Path
from polars.exceptions import PolarsPanicError
from time import time

datafolder = Path("./data")
datafolder.mkdir(parents=True, exist_ok=True)

# Copy file 2000 times
for i in range(0, 2000):
    src = Path(f"./broken.json")
    dest = datafolder / f"broken_{str(i).zfill(4)}.json"
    dest.write_text(src.read_text())

print("---- Lazy query glob ----")
stime = time()
q = pl.scan_ndjson("./data/*.json")
try:
    df = q.collect()
except PolarsPanicError as ppe:
    print(ppe)
else:
    print(df)
etime = time()
print(f"Duration: {round(etime-stime, 4)}")

print("---- Lazy query per file ----")
stime = time()
queries = []
for file in datafolder.glob("*.json"):
    q = pl.scan_ndjson(file)
    queries.append(q)
try:
    df = pl.concat(queries).collect()
except PolarsPanicError as ppe:
    print(ppe)
else:
    print(df)
etime = time()
print(f"Duration: {round(etime-stime, 4)}")


print("---- Lazy str.json_extract ----")
stime = time()
queries = []
for file in datafolder.glob("*.json"):
    q = (
        pl.LazyFrame({"json": file.read_text()})
        .with_columns(pl.col("json").str.json_extract())
        .unnest("json")
    )
    queries.append(q)
try:
    dfs = pl.collect_all(queries)
except PolarsPanicError as ppe:
    print(ppe)
else:
    df = pl.concat(dfs)
    print(df)
etime = time()
print(f"Duration: {round(etime-stime, 4)}")
```

<details>
<summary>Output</summary>

```txt
---- Lazy query glob ----
thread 'thread 'thread '<unnamed>thread 'thread 'thread 'thread 'thread '<unnamed><unnamed>' panicked at '<unnamed><unnamed><unnamed><unnamed>' panicked at '' panicked at '' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at '' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at '', ' panicked at '', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', ', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsinner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', ', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', :', :D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs::D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs267D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs267::267267::::267267::2675326753::5353:
:
5353
53note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '53
<unnamed>thread '
<unnamed>' panicked at 'thread 'thread 'thread 'thread '' panicked at '<unnamed>thread '<unnamed><unnamed><unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', thread '' panicked at '' panicked at '' panicked at '<unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs<unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at '', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:' panicked at '', ', ', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str")):D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs267inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', 267::D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:26753:267:267:53:
267:267:267
53:53:53:
53
thread '53thread '53<unnamed>
thread 'thread '<unnamed>
thread '' panicked at 'thread '<unnamed><unnamed>' panicked at '<unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed>thread '' panicked at '' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at '', thread '' panicked at '<unnamed>D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsinner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str")):<unnamed>', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', ', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', 267' panicked at 'D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsinner.add(v): ComputeError(ErrString("expected list/array in json value, got str")):267:::::', 267:26726753267267D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:53::
:::53
53535353267
:53
inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))
Duration: 0.8474

---- Lazy query per file ----
thread 'thread '<unnamed><unnamed>' panicked at 'thread 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed>', ' panicked at 'thread '' panicked at 'thread 'thread 'thread 'thread 'D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsinner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed><unnamed><unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed><unnamed>:' panicked at '' panicked at '', 267', ' panicked at '' panicked at '' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsinner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))53', :
:', ', ', ', 267D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs267D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs::::::267:267::thread '26753267267535353<unnamed>:
::
thread '535353' panicked at '<unnamed>
thread '
<unnamed>thread '' panicked at 'thread 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at 'thread '<unnamed>inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed><unnamed>', thread 'thread '<unnamed>' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))' panicked at '' panicked at 'D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', <unnamed>' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str")):D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs267inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', ', ' panicked at '::', D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsinner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))26753D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs:267267:', :   
::::535353267267267D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs
:
::thread ':
2675353<unnamed>53thread 'thread ':
' panicked at '
<unnamed>thread '' panicked at '53inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))thread '<unnamed><unnamed>' panicked at '
', inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))thread 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed>' panicked at 'thread 'D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', <unnamed>', ' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))<unnamed>:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))thread '', ', ' panicked at 'inner.add(v): ComputeError(ErrString("expected list/array in json value, got str")):D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs', <unnamed>D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rsD:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs267:267:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs' panicked at '::', 53:267:inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))267267D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs
53:267', :::
53:D:\a\polars\polars\polars\polars-io\src\ndjson_core\ndjson.rs5353267
53:
:
26753:
53
inner.add(v): ComputeError(ErrString("expected list/array in json value, got str"))
Duration: 1.1976

---- Lazy str.json_extract ----
shape: (2_000, 4)
┌───────────────────────────────────┬───────────────────────────────────┬────────────────────────────┬───────────────────┐
│ requestParameters                 ┆ plan                              ┆ debugOutput                ┆ elevationMetadata │
│ ---                               ┆ ---                               ┆ ---                        ┆ ---               │
│ struct[10]                        ┆ struct[4]                         ┆ struct[6]                  ┆ struct[2]         │
╞═══════════════════════════════════╪═══════════════════════════════════╪════════════════════════════╪═══════════════════╡
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ …                                 ┆ …                                 ┆ …                          ┆ …                 │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
└───────────────────────────────────┴───────────────────────────────────┴────────────────────────────┴───────────────────┘
Duration: 2.0188
```

</details>

**Comment by cmdlineluser** [comment]
Oops - the reason I got `InvalidEOF` instead of the `ComputeError` is because I pretty-printed the data to take a closer look at the structure and saved the buffer.

**Comment by cmdlineluser** [comment]
Just noticed this issue was linked to from SO and it now works:
```python
pl.read_json("Downloads/broken.txt")
# shape: (1, 4)
# ┌───────────────────────────────────┬───────────────────────────────────┬────────────────────────────┬───────────────────┐
# │ requestParameters                 ┆ plan                              ┆ debugOutput                ┆ elevationMetadata │
# │ ---                               ┆ ---                               ┆ ---                        ┆ ---               │
# │ struct[10]                        ┆ struct[4]                         ┆ struct[6]                  ┆ struct[2]         │
# ╞═══════════════════════════════════╪═══════════════════════════════════╪════════════════════════════╪═══════════════════╡
# │ {"04-27-2023","TRANSIT,WALK","fa… ┆ {1682609400000,{"Origin",5.0133,… ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
# └───────────────────────────────────┴───────────────────────────────────┴────────────────────────────┴───────────────────┘
```
Not sure when it was fixed.

**Comment by b-a0** [comment]
I am the "ghost" user (merged accounts) that opened this issue. I just checked if this was fixed in polars 0.19.7.

First I created a new environment:

```
conda create -n polarsdemo
conda activate polarsdemo
mamba install python
pip install polars requests
```

Then I ran below script:

```py
import polars as pl
from polars.exceptions import PolarsPanicError
import requests
import os


os.environ["RUST_BACKTRACE"] = "full"

urls = {
    "broken": "https://github.com/pola-rs/polars/files/11309421/broken.txt",
    "works": "https://github.com/pola-rs/polars/files/11309422/works.txt",
}

for fname, url in urls.items():
    print(f"---- {fname} ----")
    r = requests.get(url, allow_redirects=True)
    open(f"{fname}.json", "wb").write(r.content)
    try:
        df = pl.read_json(f"{fname}.json")
    except PolarsPanicError as ppe:
        print(ppe)
    else:
        print(df)

pl.show_versions()
```

<details>
<summary>Output</summary>

```
---- broken ----
shape: (1, 4)
┌──────────────────────────────────┬──────────────────────────────────┬────────────────────────────┬───────────────────┐
│ requestParameters                ┆ plan                             ┆ debugOutput                ┆ elevationMetadata │
│ ---                              ┆ ---                              ┆ ---                        ┆ ---               │
│ struct[10]                       ┆ struct[4]                        ┆ struct[6]                  ┆ struct[2]         │
╞══════════════════════════════════╪══════════════════════════════════╪════════════════════════════╪═══════════════════╡
│ {"04-27-2023","TRANSIT,WALK","fa ┆ {1682609400000,{"Origin",5.0133, ┆ {99,123,[122],2,224,false} ┆ {47.694914,false} │
│ …                                ┆ …                                ┆                            ┆                   │
└──────────────────────────────────┴──────────────────────────────────┴────────────────────────────┴───────────────────┘
---- works ----
shape: (1, 4)
┌──────────────────────────────────┬──────────────────────────────────┬────────────────────────────┬───────────────────┐
│ requestParameters                ┆ plan                             ┆ debugOutput                ┆ elevationMetadata │
│ ---                              ┆ ---                              ┆ ---                        ┆ ---               │
│ struct[10]                       ┆ struct[4]                        ┆ struct[6]                  ┆ struct[2]         │
╞══════════════════════════════════╪══════════════════════════════════╪════════════════════════════╪═══════════════════╡
│ {"04-27-2023","TRANSIT,WALK","fa ┆ {1682576400000,{"Origin",5.8045, ┆ {99,327,[326],2,428,false} ┆ {47.694914,false} │
│ …                                ┆ …                                ┆                            ┆                   │
└──────────────────────────────────┴──────────────────────────────────┴────────────────────────────┴───────────────────┘
--------Version info---------
Polars:              0.19.7
Index type:          UInt32
Platform:            Windows-10-10.0.19045-SP0
Python:              3.12.0 | packaged by conda-forge | (main, Oct  3 2023, 08:26:13) [MSC v.1935 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
gevent:              <not installed>
matplotlib:          <not installed>
numpy:               <not installed>
openpyxl:            <not installed>
pandas:              <not installed>
pyarrow:             <not installed>
pydantic:            <not installed>
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

Note that I now use `read_json` instead of `read_ndjson` as the former no longer has the restriction that the JSON file has to be an array. Can't find since what version that has changed, but it works, so this issue can be closed (can't do it myself due to the ghost user).
